src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -282,6 +282,34 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
+  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = workflowLoggerWithCall(backendCall)
+    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
+      Try(gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log))) match {
+        case Success(attempts) if attempts.forall(_.isSuccess) =>
+          postProcess(backendCall) match {
+            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
+            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
+              // TODO: What can we return here to retry this operation?
+              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
+              log.error(error, ex)
+              FailedExecutionHandle(new Throwable(error, ex))
+            case Failure(ex) => FailedExecutionHandle(ex)
+          }
+        case Success(attempts) =>
+          TryUtil.sequence(attempts.toSeq) match {
```

Comment:
I'm not convinced by arguments like "other people do it"... Also our "homebrew" code is 2 lines, and 2 lines which I argue are pretty hard to mess up.  I'm looking at the code and the `scalaz` version makes you do this:

```scala
val validationNel = nels.toList.sequence[({type λ[a] = ValidationNel[String, a]})#λ, StorageObject]
validationNel.leftMap("Could not copy files: " + _.list.mkString(", ")).toValidationNel
```

Whereas we could just replace that with

```scala
TryUtil.sequence(list)
```

And the latter way will make the return type `Try[Seq[StorageObject]]` as opposed to a `ValidationNEL[String, Seq[StorageObject]]`.  I don't know about you, but I'd *much* rather have a `Try` return type than a `ValidationNEL` if possible.